src/TokenFuzzChecks.sol

```diff
@@ -0,0 +1,448 @@
+    // Mint/Burn
+    // ************************************************************************************************************
+
+    function fuzzBulkCheckMintBurn(
+        address _token,
+        string memory _contractName,
+        address who,
+        uint256 mintAmount,
+        uint256 burnAmount
+    ) internal {
+        fuzzCheckMint(_token, _contractName, who, mintAmount);
+        fuzzCheckBurn(_token, who, mintAmount, burnAmount);
+        fuzzCheckBurnInsufficientBalance(_token, _contractName, who, mintAmount, burnAmount);
+        fuzzCheckTokenModifiers(_token, _contractName, who);
+    }
+
+    function fuzzCheckMint(
+        address _token,
+        string memory _contractName,
+        address to,
+        uint256 mintAmount
+    ) internal {
+        uint256 prevSupply = TokenLike(_token).totalSupply();
+        uint256 prevToBalance = TokenLike(_token).balanceOf(to);
+        mintAmount = bound(mintAmount, 0, type(uint256).max - prevSupply);
+        uint256 prevWard = TokenLike(_token).wards(address(this));
+        _token.setWard(address(this), 1);
+        if (to != address(0) && to != _token) {
+            vm.expectEmit(true, true, true, true);
+            emit Transfer(address(0), to, mintAmount);
+        } else {
+            vm.expectRevert(abi.encodePacked(_contractName, "/invalid-address"));
+        }
+
+        TokenLike(_token).mint(to, mintAmount);
+
+        if (to != address(0) && to != _token) {
+            assertEq(TokenLike(_token).totalSupply(), prevSupply + mintAmount);
+            assertEq(TokenLike(_token).balanceOf(to), prevToBalance + mintAmount);
+        }
+        _token.setWard(address(this), prevWard);
+    }
+
+    function fuzzCheckBurn(
+        address _token,
+        address from,
+        uint256 mintAmount,
+        uint256 burnAmount
+    ) internal {
+        vm.assume(from != address(0) && from != _token);
+        uint256 prevSupply = TokenLike(_token).totalSupply();
+        uint256 prevFromBalance = TokenLike(_token).balanceOf(from);
+        mintAmount = bound(mintAmount, 0, type(uint256).max - prevSupply);
+        burnAmount = bound(burnAmount, 0, prevFromBalance + mintAmount);
+        deal(_token, from, prevFromBalance + mintAmount, true);
+
+        vm.expectEmit(true, true, true, true);
+        emit Transfer(from, address(0), burnAmount);
+        vm.prank(from); TokenLike(_token).burn(from, burnAmount);
+
+        assertEq(TokenLike(_token).totalSupply(), prevSupply + mintAmount - burnAmount);
+        assertEq(TokenLike(_token).balanceOf(from), prevFromBalance + mintAmount - burnAmount);
+    }
+
+    function fuzzCheckBurnInsufficientBalance(
+        address _token,
+        string memory _contractName,
+        address to,
+        uint256 mintAmount,
+        uint256 burnAmount
+    ) internal {
+        uint256 prevSupply = TokenLike(_token).totalSupply();
+        if(prevSupply == type(uint256).max) return;
```

Comment:
```suggestion
        if (prevSupply == type(uint256).max) return;
```

Reply:
or maybe better:
```suggestion
        vm.assume(prevSupply <type(uint256).max);
```

Reply:
I kept it as a return here because I'm not entirely sure `vm.assume` should be used on variables indirectly derived from the fuzz inputs. It may be fine though. But the other thing is that just looking at this test, it's not immediately obvious that we are excluding a narrow range here. It may look like it but strictly speaking we don't know that without also looking at what other tests are doing with the total supply.

Reply:
Yes, clear!